chirp/drivers/radtel_rt493.py

```diff
@@ -0,0 +1,1152 @@
+        time.sleep(0.002)
+    except Exception:
+        raise errors.RadioError("Error sending data to radio")
+
+
+def _recv(radio, addr, length):
+    """Get data from the radio """
+    hdr = _rawrecv(radio, 4)  # Read 4 byte header
+    data = _rawrecv(radio, length)  # Read data
+
+    # DEBUG
+    LOG.info("Response:")
+    LOG.debug(util.hexprint(hdr + data))
+
+    c, a, resp_length = struct.unpack(">BHB", hdr)
+    if a != int.from_bytes(addr, byteorder="big") \
+            or resp_length != length or c != ord("W"):
+        LOG.error("Invalid answer for block 0x%04s:" % util.hexprint(addr))
+        LOG.debug("CMD: %s  ADDR: %04x  SIZE: %02x" % (c, a, resp_length))
+        raise errors.RadioError("Unknown response from the radio")
+
+    return data
+
+
+def _enter_programming_mode(radio):
+    _clean_buffer(radio)
+
+    _rawsend(radio, radio._magic)
+    LOG.debug("Sent magic sequence")
+    ack = _rawrecv(radio, 2)
+    LOG.debug("Received magic sequence response")
+    if len(ack) != 2 or ack[1:2] != COMMAND_ACCEPT:
+        if ack:
+            LOG.error("Received: Len=%i Data=%s"
+                      % (len(ack), util.hexprint(ack)))
+        raise errors.RadioError("Radio refused to enter programming mode")
+
+    radio.pipe.reset_input_buffer()
+    _rawsend(radio, b"\x02")
+    ident = _rawrecv(radio, radio._magic_response_length)
+
+    if not ident.startswith(radio._fingerprint):
+        raise errors.RadioError("Radio returned unknown identification string")
+
+    LOG.info("Radio entered programming mode")
+    LOG.debug("Radio identification: %s" % util.hexprint(ident))
+
+    _rawsend(radio, COMMAND_ACCEPT)
+    ack = _rawrecv(radio, 1)
+    if ack != COMMAND_ACCEPT:
+        if ack:
+            LOG.error("Got %s" % util.hexprint(ack))
+        raise errors.RadioError("Radio refused to enter programming mode")
+
+    return ident
+
+
+def _exit_programming_mode(radio):
+    try:
+        radio.pipe.write(b"\x45")
+    except Exception:
+        raise errors.RadioError("Radio refused to exit programming mode")
+
+
+def _get_memory_address(channel_id):
+    block = (channel_id - 1) // 16           # Determines high byte
+    offset = ((channel_id - 1) % 16) * 0x10  # Determines low byte
+
+    return (block, offset)
+
+
+def _read_block(radio, channel_id):
+    address = bytes(_get_memory_address(channel_id))
+
+    try:
+        # 52=read | 20 = hex(32 bytes) = length of 2 channels
+        # + settings without header
+        cmd = b"\x52" + address + b"\x20"
+        _rawsend(radio, cmd)
+        data = _recv(radio, address, 32)
+
+        if address != b"\x00\x00":
+            _rawsend(radio, COMMAND_ACCEPT)
+            response = _rawrecv(radio, 1)
+            if response != COMMAND_ACCEPT:
+                raise errors.RadioError("Radio refused to read block at %04s"
+                                        % util.hexprint(address))
+
+        return data
+    except Exception:
```

Comment:
Again, you're swallowing your specific error at L162 and re-raising here, as well as any of the specific things that `_rawsend` and `_rawrecv` are raising, like L119 which would be quite important.